src/main/java/com/thebois/views/game/ViewableResource.java

```diff
@@ -0,0 +1,69 @@
+package com.thebois.views.game;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.Texture;
+
+import com.thebois.Pawntastic;
+import com.thebois.models.world.resources.ResourceType;
+import com.thebois.views.TextureUtils;
+
+/**
+ * Viewable Resource that correspond to TerrainType but has view related information.
+ */
+public enum ViewableResource {
+    /**
+     * Viewable Water tile.
+     */
+    WATER(
+        ResourceType.WATER,
+        TextureUtils.createSquareTexture((int) Pawntastic.TILE_SIZE, Color.valueOf("#1E90FF"))),
+    /**
+     * Viewable Tree tile.
+     */
+    TREE(
+        ResourceType.TREE,
+        TextureUtils.createTriangleTexture((int) Pawntastic.TILE_SIZE, Color.valueOf("#006400"))),
+    /**
+     * Viewable Rock tile.
+     */
+    ROCK(
+        ResourceType.ROCK,
+        TextureUtils.createTriangleTexture((int) Pawntastic.TILE_SIZE, Color.valueOf("#808080")));
+    private static final Map<ResourceType, ViewableResource> BY_TYPE = new HashMap<>();
+
+    static {
+        for (final ViewableResource resource : ViewableResource.values()) {
+            BY_TYPE.put(resource.getType(), resource);
+        }
+    }
+
+    private final Texture texture;
+    private final ResourceType type;
+
+    ViewableResource(final ResourceType type, final Texture texture) {
+        this.texture = texture;
+        this.type = type;
+    }
+
+    /**
+     * Get ViewableResource with given ResourceType.
```

Comment:
style

I like to write javadocs using the words that the types represent rather than the types themselves. And I try to use synonyms to give a better understanding of what it does. I do this because I can see from the method signature that it returns a `ViewableResource` and takes in a `ResourceType`, you don't need to repeat that in the javadoc.

I'd first convert it like this:
"Gets a viewable resource associated with a given resource type."

Then I'd use synonyms instead, to give another perspective (other than what the reader already can infer from just the type names alone):
"Gets the graphical representation of a given type of resource."